File: fc-chat/demo/part3/ch4/message-system-add-push/src/main/java/net/prostars/messagesystem/service/PushService.java
package net.prostars.messagesystem.service;

import java.util.HashSet;
import net.prostars.messagesystem.dto.domain.UserId;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

@Service
public class PushService {

  private static final Logger log = LoggerFactory.getLogger(PushService.class);

  private final HashSet<String> pushMessageTypes = new HashSet<>();

  public void registerPushMessageType(String messageType) {
    pushMessageTypes.add(messageType);
  }

  public void pushMessage(UserId userId, String messageType, String message) {
    if (pushMessageTypes.contains(messageType)) {
      log.info("push message: {} to user: {}", message, userId);
    }
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/service/PushService.kt
package com.yscorp.withpush.messagesystem.service

import net.prostars.messagesystem.dto.domain.UserId

@org.springframework.stereotype.Service
class PushService {
    private val pushMessageTypes = java.util.HashSet<String>()

    fun registerPushMessageType(messageType: String) {
        pushMessageTypes.add(messageType)
    }

    fun pushMessage(userId: UserId?, messageType: String, message: String?) {
        if (pushMessageTypes.contains(messageType)) {
            log.info("push message: {} to user: {}", message, userId)
        }
    }

    companion object {
        private val log: org.slf4j.Logger = org.slf4j.LoggerFactory.getLogger(PushService::class.java)
    }
}
